talks/views.py

```diff
@@ -20,4 +21,29 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['talks'] = Talk.objects.all()[:3]
         context['follow_talks'] = FollowTalk.objects.all()
-        return context
\ No newline at end of file
+        return context
+
+def talkRequest(request):
+    """Takes a user request to find a talk and it returns a json object
+    with talk names if the match exists or it returns None as json object"""
+
+    user_input = request.GET.get('talk-name-input', None)
+    if not user_input:
+        data = {'Matched Talks': None}
+        return JsonResponse(data)
+    else:     
+        talks = Talk.objects.filter(name__istartswith=f'{user_input}')
+        
+        if talks:
+            talk_names = [x.name for x in talks]
```

Comment:
This is extremely inefficient, use a `values_list` in the `talks` query above. At this point the `talks` query has already executed.
```python
talks = Talk.objects.filter(name__icontains=f'{user_input}').values_list('name', flat=True)
```